drivers/adc/adc_tla2021.c

```diff
@@ -197,7 +200,9 @@ static void tla2021_perform_read(const struct device *dev)
 	/*
 	 * Wait until sampling is done
 	 */
+	k_usleep(WAKEUP_TIME_US + CONVERSION_TIME_US);
 	do {
+		k_yield();
```

Comment:
I would move the yield down to right before the condition to not have a sleep (yield) followed by another yield:
```
        k_yield();
} while (!(reg & REG_CONFIG_OS_msk));
```